pages/6_📖_TOPP-Workflow_Docs.py

```diff
@@ -0,0 +1,434 @@
+from src.common import page_setup
+
+page_setup()
+
+wf = Workflow()
+
+st.title("📖 Workflow Framework Docs")
+
+st.markdown(
+"""
+## Features
+
+- streamlined methods for uploading files, setting parameters, and executing workflows
+- automatic parameter handling
+- quickly build parameter interface for TOPP tools with all parameters from *ini* files
+- automatically create a log file for each workflow run with stdout and stderr
+- workflow output updates automatically in short intervalls
+- user can leave the app and return to the running workflow at any time
+- quickly build a workflow with multiple steps channelling files between steps
+#
+"""
+)
+
+with st.expander("**Example User Interface**", True):
+    t =  st.tabs(["📁 **File Upload**", "⚙️ **Parameter Settings**", "🚀 **Execution**", "📊 **Results**"])
+    with t[0]:
+        wf.show_file_upload_section()
+
+    with t[1]:
+        wf.show_parameter_section()
+
+    with t[2]:
+        wf.show_execution_section()
+        
+    with t[3]:
+        wf.show_results_section()
+
+st.markdown(
+"""
+## Quickstart
+
+This repository contains a module in `src/workflow` that provides a framework for building and running analysis workflows.
+
+The `WorkflowManager` class provides the core workflow logic. It uses the `Logger`, `Files`, `DirectoryManager`, `ParameterManager`, and `CommandExecutor` classes to setup a complete workflow logic.
+
+To build your own workflow edit the file `src/TOPPWorkflow.py`. Use any streamlit components such as tabs (as shown in example), columns, or even expanders to organize the helper functions for displaying file upload and parameter widgets.
+
+> 💡 Simply set a name for the workflow and overwrite the **`upload`**, **`parameter`**, **`execution`** and **`results`** methods in your **`Workflow`** class.
+
+The file `pages/6_TOPP-Workflow.py` displays the workflow content and can, but does not have to be modified.
+
+The `Workflow` class contains four important members, which you can use to build your own workflow:
+
+> **`self.params`:** dictionary of parameters stored in a JSON file in the workflow directory. Parameter handling is done automatically. Default values are defined in input widgets and non-default values are stored in the JSON file.
+
+> **`self.ui`:** object of type `StreamlitUI` contains helper functions for building the parameter and file upload widgets.
+
+> **`self.executor`:** object of type `CommandExecutor` can be used to run any command line tool alone or in parallel and includes a convenient method for running TOPP tools.
+
+> **`self.logger`:** object of type `Logger` to write any output to a log file during workflow execution.
+
+Handling input and output files in the `Workflow.execution` method for processes is done with the `Files` class, handling file types and creation of output directories.
+"""
+)
+
+with st.expander("**Complete example for custom Workflow class**", expanded=False):
+    st.code(
+"""
```

Comment:
How about using a library to retrieve the code? It's better than copy and paste whenever something changes in the corresponding codes
https://docs.python.org/3/library/inspect.html#inspect.getsource

Reply:
Amazing! Way better.